### PR TITLE
Add an example for `thrift` tutorial

### PR DESCRIPTION
**Pull request by jrhee17**
Motivation:

This change adds a `examples:tutorials:thrift` module, which is a part of the series of tutorials.

Modifications:

- Add an `examples:tutorials:thrift`
- Add a `BlogServiceImpl` thrift service implementation.
- Add a `BlogServiceExceptionHandler` which illustrates how exceptions can be translated using an `ExceptionHandler`.
- Add unit tests to demonstrate client behavior.

Result:

- Preparation is done to add a `thrift` tutorial

### CODE REVIEW

File: examples/tutorials/thrift/src/main/thrift/blog.thrift [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+struct BlogPost {
+    1: i32 id;
+    2: string title;
+    3: string content;
+    4: i64 createdAt;
+    5: i64 modifiedAt;
+}
+
+exception BlogNotFoundException {
+    1: required string reason
```
Comment (minwoox) [inline examples/tutorials/thrift/src/main/thrift/blog.thrift:42]: Could you share your rationale behind using `required`? 😄

File: examples/tutorials/thrift/src/main/thrift/blog.thrift [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+* My first BlogService in Armeria!
+**/
+service BlogService {
+
+  /**
+  * Creates a blog post.
+  **/
+  BlogPost createBlogPost(1:CreateBlogPostRequest request),
```
Comment (minwoox) [inline examples/tutorials/thrift/src/main/thrift/blog.thrift:53]: The indentation is different from the structs above.

File: examples/tutorials/thrift/src/test/java/example/armeria/server/blog/thrift/BlogServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+    @Test
+    @Order(1)
+    void createBlogPost() throws TException {
+        final CreateBlogPostRequest request = new CreateBlogPostRequest()
+                .setTitle("My first blog")
+                .setContent("Hello Armeria!");
+        final BlogPost response = client.createBlogPost(request);
+        assertThat(response.getTitle()).isEqualTo(request.getTitle());
+        assertThat(response.getContent()).isEqualTo(request.getContent());
```
Comment (minwoox) [inline examples/tutorials/thrift/src/test/java/example/armeria/server/blog/thrift/BlogServiceTest.java:64]: How about also checking `id`?

**Review by minwoox — COMMENTED** [review]
Thanks! Left small suggestions. 😄

File: examples/tutorials/thrift/src/test/java/example/armeria/server/blog/thrift/BlogServiceTest.java (hunk 3 above)
Comment (jrhee17) [reply examples/tutorials/thrift/src/test/java/example/armeria/server/blog/thrift/BlogServiceTest.java:65]: added an assertion to check `>= 0` 🙇

File: examples/tutorials/thrift/src/main/thrift/blog.thrift (hunk 1 above)
Comment (jrhee17) [reply examples/tutorials/thrift/src/main/thrift/blog.thrift]: There was none 😅   removed

File: examples/tutorials/thrift/src/main/java/example/armeria/server/blog/thrift/BlogServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        final BlogPost stored = updated;
+        resultHandler.onComplete(stored);
+    }
+
+    @Override
+    public void getBlogPost(GetBlogPostRequest request, AsyncMethodCallback<BlogPost> resultHandler)
+            throws TException {
+        final BlogPost blogPost = blogPosts.get(request.getId());
+        if (blogPost == null) {
+            // throw new BlogNotFoundException("The blog post does not exist. ID: " + request.getId());
```
[the remark this answers is not in the document]
Comment (jrhee17) [reply examples/tutorials/thrift/src/main/java/example/armeria/server/blog/thrift/BlogServiceImpl.java:53]: removed 😅

File: examples/tutorials/thrift/src/main/java/example/armeria/server/blog/thrift/BlogServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+    @Override
+    public void deleteBlogPost(DeleteBlogPostRequest request, AsyncMethodCallback<Void> resultHandler)
+            throws TException {
+        final BlogPost removed = blogPosts.remove(request.getId());
+        if (removed == null) {
+            // throwing an exception will also have the same effect
+            // throw new NullPointerException("The blog post does not exist. ID: " + request.getId());
+            resultHandler.onError(
+                    new NullPointerException("The blog post does not exist. ID: " + request.getId()));
```
[the remark this answers is not in the document]
Comment (jrhee17) [reply examples/tutorials/thrift/src/main/java/example/armeria/server/blog/thrift/BlogServiceImpl.java:99]: If I throw a `BlogNotFoundException` here, then I guess there is no way to test the behavior of `BlogServiceExceptionHandler`.
Do you think it's better to just remove `BlogServiceExceptionHandler`?

[the remark this answers is not in the document]
Comment (minwoox) [reply examples/tutorials/thrift/src/main/java/example/armeria/server/blog/thrift/BlogServiceImpl.java:99]: > Do you think it's better to just remove BlogServiceExceptionHandler

I think we need that. 😄 
How about using `IllegalArgumentException` as we do in other tutorials?

File: examples/tutorials/thrift/src/main/thrift/blog.thrift [hunk 6]
```diff
@@ -0,0 +1,95 @@
+namespace java example.armeria.blog.thrift
+
+/**
+ * A request object for creating a blog post.
+ **/
```
Comment (minwoox) [inline examples/tutorials/thrift/src/main/thrift/blog.thrift:5]: nit:
```suggestion
 */
```

Comment (jrhee17) [reply examples/tutorials/thrift/src/main/thrift/blog.thrift]: Oops 😅  Sorry about this. Fixed

File: examples/tutorials/thrift/src/main/java/example/armeria/server/blog/thrift/BlogServiceImpl.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (jrhee17) [reply examples/tutorials/thrift/src/main/java/example/armeria/server/blog/thrift/BlogServiceImpl.java]: > How about using the same exception?

I realized you meant the same exception as the other tutorials 😅  Fixed to use `IllegalArgumentException`

**Review by minwoox — APPROVED** [review]
Let's contact to geun nim. 🤣

**Review by ikhoon — APPROVED** [review]
Looks good to me if the `java` flag is migrated to the brand new `java11`.

**Pull request merged**